pyleecan/Methods/Output/LUTdq/interp_Ploss_dqh.py

```diff
@@ -33,23 +35,15 @@ def interp_Ploss_dqh(self, Id, Iq, N0):
 
     felec = N0 / 60 * p
 
-    type_skin_effect = self.simu.loss.type_skin_effect
-    Tsta = self.simu.loss.Tsta
-
-    Ploss_dqh = np.zeros((len(self.output_list), 5))
+    Ploss_dqh = np.zeros(
+        (len(self.output_list), len(self.output_list[0].loss.loss_list))
+    )
     for ii, out in enumerate(self.output_list):
         OP = out.elec.OP.copy()
         OP.felec = felec
-        Ploss_dqh[ii, 0] = comp_loss_joule(
-            lam=self.simu.machine.stator,
-            OP=OP,
-            T_op=Tsta,
-            type_skin_effect=type_skin_effect,
-        )
-        Ploss_dqh[ii, 1] = out.loss.get_loss_group("stator core", felec)
-        Ploss_dqh[ii, 2] = out.loss.get_loss_group("rotor magnets", felec)
-        Ploss_dqh[ii, 3] = out.loss.get_loss_group("rotor core", felec)
-        Ploss_dqh[ii, 4] = out.loss.get_loss_group("stator winding", felec)
+        for kk, loss in enumerate(out.loss.loss_list):
+            if loss.loss_model not in exclude_models:
+                Ploss_dqh[ii, kk] = loss.get_loss_scalar(felec)
 
     # Get unique Id, Iq sorted in ascending order
     OP_matrix = self.get_OP_array("N0", "Id", "Iq")
```

Comment:
Regrading 2D interpolation's 'if' branch: Will jd ad jq from np.unique ever be the valid indices. Or is it possible that we access the wrong OP in the interpolation, if the OP_matrix is odd somehow?
Maybe if we should adapt 'elif' branch indexing here.

Actual interpolation (from line 87): You are using linear interpolation. Have you tried (bi)cubic as well? Maybe there could be an option somewhere.